### PR TITLE
fix(telemetry docs): replace p&t concept doc with observability doc

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR replaces the existing concept guide for *Policies and Telemetry* with a doc on *Observability*.

The previous doc focused on Mixer and how it works. The new guide attempts to highlight the observability features of Istio at a high-level, without focusing on the implementation details. Policy details are removed to avoid confusion (they are also covered in the *Security* concept guide).

The old `Concepts > Policies and Telemetry` content is moved to the reference section and given a new name: `Reference > Configuration > Policies and Telemetry > Mixer Configuration Model`. The *Reliability and latency * section is removed (as this information is outdated and now covered in the perf-and-scalability sections of the docs).

The *Default Metrics* documentation is also updated to reflect Prometheus metrics names, to make it easier to consume.

NOTE: The *Control Plane Metrics* subsection of the new doc does not currently include links to the documentation on those metrics, as that has not been generated. I plan on working on that in a separate PR.

### CODE REVIEW

File: content/docs/concepts/observability/index.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  source_workload_namespace="default"
+} 214
+```
+
+### Control Plane Metrics
+
+Each Istio component (Pilot, Galley, Mixer) also provides a collection of self-monitoring metrics. These metrics allow monitoring of the behavior
+of Istio itself (as distinct from that of the services within the mesh).
+
+For more information on which metrics are maintained, please the reference docs for each of the components:
```
Comment (bianpengyuan) [inline content/docs/concepts/observability/index.md]: Please see?

File: content/docs/concepts/observability/index.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+- Pilot
```
Comment (bianpengyuan) [inline content/docs/concepts/observability/index.md]: Should these be link? Also should citadel be added here?

File: content/docs/concepts/observability/index.md (hunk 1 above)
Comment (geeknoid) [reply content/docs/concepts/observability/index.md]: Actually, we're not that polite in our docs. That should just be "see the reference docs"

File: content/docs/concepts/observability/index.md (hunk 2 above)
Comment (douglas-reid) [reply content/docs/concepts/observability/index.md]: yes, but the target of those links doesn't exist yet.

**Comment by douglas-reid** [comment]
closing in favor of #4448

**Pull request closed without merging**